jsx/Form.js

```diff
@@ -1285,6 +1285,95 @@ LinkElement.defaultProps = {
   href: null
 };
 
+/**
+ * Checkbox Component
+ * React wrapper for a <input type="checkbox"> element.
+ */
+class CheckboxElement extends React.Component {
+  constructor() {
+    super();
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(e) {
+    this.props.onUserInput(this.props.name, e.target.checked);
+  }
+
+  render() {
+    let disabled = this.props.disabled ? 'disabled' : null;
+    let required = this.props.required ? 'required' : null;
+    let errorMessage = null;
+    let requiredHTML = null;
+    let elementClass = 'row form-group';
+    let label = null;
+
+    // Add required asterix
+    if (required) {
+      requiredHTML = <span className="text-danger">*</span>;
+    }
+
+    // Add error message
+    if (this.props.errorMessage) {
+      errorMessage = <span>{this.props.errorMessage}</span>;
+      elementClass = 'row form-group has-error';
+    }
+
+    if (this.props.label) {
```

Comment:
This was introduced for potential in-line formatting, in the case where there was no label. However, it is not necessary and label can be made a mandatory field.

Reply:
i agree with label not needing to be a mandatory field